### PR TITLE
docs: Gpmt tool reference

### PR DESCRIPTION
**Pull request by mperezfuster**
This PR replaces the Support KB article: https://community.pivotal.io/s/article/Greenplum-Magic-Tool-GPMT?language=en_US and the 4 linked articles from that one (some of them internal).

Preview: https://mireia-gpmt.sc2-04-pcf1-apps.oc.vmware.com/7-0/utility_guide/ref/gpmt.html

### CODE REVIEW

File: gpdb-doc/dita/utility_guide/ref/gpmt.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+    <section>
+      <title>Tools</title>
+      <parml>
+        <plentry>
+          <pt><xref href="gpmt-analyze_session.xml" type="topic" format="dita"></xref></pt>
+          <pd>Collect information from a hung GPDB session for remote analysis.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>catalogbackup</varname></pt>
+          <pd>For VMware Support use only. Backup catalog prior to performing catalog repairs.</pd>
```
Comment (VMWareJennyBeth) [inline gpdb-doc/dita/utility_guide/ref/gpmt.xml:24]: Backup is one word when a noun. When a verb, use two: back up

File: gpdb-doc/dita/utility_guide/ref/gpmt.xml [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+          <pt><xref href="gpmt-gpstatscheck.xml" type="topic" format="dita"></xref></pt>
+          <pd>Check for missing stats on objects used in a query.</pd>
+        </plentry>
+        <plentry>
+          <pt><xref href="gpmt-packcore.xml" type="topic" format="dita"></xref></pt>
+          <pd>Package core files into single tarball for remote analysis.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>primarymirror_lengths</varname></pt>
+          <pd>For VMware Support use only. Check if primary and mirror AO and AOCO relfiles are the
```
Comment (VMWareJennyBeth) [inline gpdb-doc/dita/utility_guide/ref/gpmt.xml:49]: Check whether

File: gpdb-doc/dita/utility_guide/ref/gpmt.xml [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    <section>
+      <title>Options</title>
+      <parml>
+        <plentry>
+          <pt><varname>hostfile</varname></pt>
+          <pd>Generate hostfiles for use with other tools.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>replcheck</varname></pt>
+          <pd>Check if tool is replicated to all hosts.</pd>
```
Comment (VMWareJennyBeth) [inline gpdb-doc/dita/utility_guide/ref/gpmt.xml:68]: Check whether

**Review by VMWareJennyBeth — CHANGES_REQUESTED** [review]
Looks really good, Mireia. I had just a handful of suggestions.

File: gpdb-doc/dita/utility_guide/ref/gpmt.xml [hunk 4]
```diff
@@ -0,0 +1,94 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic PUBLIC "-//OASIS//DTD DITA Topic//EN" "topic.dtd">
+<topic id="topic_vzy_mkx_bpb">
+  <title>gpmt</title>
+  <body>
+    <p> GPMT (Greenplum Magic Tool) provides a set of diagnostic utilities to troubleshoot and
+      resolve common supportability issues, along with a consistent method for gathering information
+      required by VMware Support.</p>
+    <section>
```
Comment (lenapivotal) [inline gpdb-doc/dita/utility_guide/ref/gpmt.xml:9]: Have we agreed to add a VMware Support tool to the Greenplum docs? I find that a bit confusing. Who supports it and who owns it?

Comment (mperezfuster) [reply gpdb-doc/dita/utility_guide/ref/gpmt.xml:9]: Yes GPMT will be included with GPDB 5.x and 6.x and the KB article will be made private that's why Engineering have requested to have this documented within the docs. I am not sure but I presume they will own it from now on.

File: gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml [hunk 5]
```diff
@@ -0,0 +1,180 @@
...
+            logs will be collected.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>-free-space </varname></pt>
+          <pd>Free space threshold which will exit log collection if reached. Default value is
+            10%.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>-c</varname></pt>
+          <pd>Comma seperated list of content IDs to collect logs from.</pd>
```
[the remark this answers is not in the document]
Comment (dyozie) [reply gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml:32]: Also global change:

to collect logs from -> from which logs are collected

File: gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml [hunk 6]
```diff
@@ -0,0 +1,180 @@
...
+          <pd>Working directory (defaults to current directory).</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>-segdir</varname></pt>
+          <pd>Segment temporary directory (defaults to /tmp).</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>-skip-master</varname></pt>
+          <pd>When running gp_log_collector, the generated tarball can be very large. Use this
+            option to skip Greenplum Master log collection when only Greenplum Segment logs are
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml:65]: Master -> master segment
Segment -> segment

File: gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml [hunk 7]
```diff
@@ -0,0 +1,180 @@
...
+          <pd>Collect all GPText logs along with the Greenplum logs.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>-with-gptext-only</varname>
+          </pt>
+          <pd>Only Collect GPText logs.</pd>
+        </plentry>
+      </parml>
+    </section>
+    <p>
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml:79]: Make this an action Note element, and nest it in the --hostfile plentry above

File: gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml [hunk 8]
```diff
@@ -0,0 +1,180 @@
...
+              </ul>
+            </entry>
+          </row>
+          <row>
+            <entry>Database related files from all hosts</entry>
+            <entry>
+              <ul id="ul_e45_bzx_bpb">
+                <li><codeph>$SEG_DIR/pg_hba.conf</codeph></li>
+                <li><codeph>$SEG_DIR/pg_log/</codeph></li>
+                <li><codeph>$SEG_DIRE/postgresql.conf</codeph></li>
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml:142]: SEG_DIRE -> SEG_DIR

File: gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml [hunk 9]
```diff
@@ -0,0 +1,180 @@
...
+                <li><codeph>commands/bash/-c_echo $PATH</codeph></li>
+                <li><codeph>commands/bash/-c_ps -ef | grep solr</codeph></li>
+                <li><codeph>commands/bash/-c_ps -ef | grep zookeeper</codeph></li>
+              </ul>
+            </entry>
+          </row>
+        </tbody>
+      </tgroup>
+    </table>
+    <p>Note: some commands might not be able to be run if user does not have enough permission.</p>
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-gp_log_collector.xml:166]: Make into Note element, and reword to:

Some commands may require additional permission to run.

Also relocate this note to the table cell that lists the various commands.

File: gpdb-doc/dita/utility_guide/ref/gpmt-gpstatscheck.xml [hunk 10]
```diff
@@ -0,0 +1,44 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1"><title id="oz137116">gpmt gpstatscheck</title><body>
+    <p>This tool can be used to verify if all tables involved in a query have optimal statistics.
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-gpstatscheck.xml:5]: can be used to verify if all -> helps to verify that

File: gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml [hunk 11]
```diff
@@ -0,0 +1,52 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!DOCTYPE topic
+  PUBLIC "-//OASIS//DTD DITA Composite//EN" "ditabase.dtd">
+<topic id="topic1">
+  <title id="oz137116">gpmt packcore</title>
+  <body>
+    <p>This tool collects all the necessary files and generates a script to run the
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml:7]: to run the \<codeph>gdb\</codeph> command that makes -> that runs the \<codeph>gdb\</codeph> command to create

File: gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+        <codeph>gdb</codeph> command that makes a core file readable by an external host. It is used
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml:8]: It is used -> Use this command

File: gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+      when VMware Support needs to collect core files generated by the Greenplum processes.</p>
+    <section>
+      <title>Usage</title>
+      <codeblock><b>gpmt</b> <b>packcore</b> -cmd COMMAND -core COREFILE [<varname>-binary BINARY </varname>]
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml:12]: Not sure what COMMAND  is referring to here. There should be a -cmd description in the options that are described to indicate what commands are available & what they do. If only `collect` is what they'd be doing, just add that as part of the required syntax.

File: gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml [hunk 14]
```diff
@@ -0,0 +1,52 @@
...
+          <pt><varname>-core</varname></pt>
+          <pd>Corefile name.</pd>
+        </plentry>
+        <plentry>
+          <pt><varname>-binary </varname></pt>
+          <pd>Binary name that generated the core file. </pd>
+        </plentry>
+        <plentry>
+          <pt><varname>-keep_tmp_dir</varname></pt>
+          <pd>Do not delete temp directory.</pd>
```
Comment (dyozie) [inline gpdb-doc/dita/utility_guide/ref/gpmt-packcore.xml:37]: temp -> the temporary

**Review by dyozie — CHANGES_REQUESTED** [review]

**Review by dyozie — APPROVED** [review]

**Pull request merged**